src/cc/lib/graph/metadata.h

```diff
@@ -19,6 +19,10 @@ struct Metadata
     explicit Metadata(std::filesystem::path path, std::string config_path = "");
     void Write(std::filesystem::path path) const;
 
+    // Optional graph information.
+    std::string m_name = "";
+    std::string m_data_path = "";
```

Comment:
Server already accepts [a data location to load graph](https://github.com/microsoft/DeepGNN/blob/13f0b009cc78f147c8cb623e8205317e48aa7fe9/src/python/deepgnn/graph_engine/snark/server.py#L49) from a different location other than meta file. How conflicts are going to be resolved?

Reply:
There will not be conflicts, we will only use meta.data_path if server is in active_monitor mode and it detects a meta.json in the monitored folder. But I will make it so the meta file supports the list of partition data dirs instead of just a single one though

Reply:
Please, try to avoid multiple config entries for data sources. I.e. if server is in active_monitor state and was started with data locations specified from command line, then we are going to monitor a new location specified in data_path?